apps/api/src/server/services/UserService.ts

```diff
@@ -85,4 +181,63 @@ export class UserService {
       return new Error('Internal Server Error');
     }
   }
+
+  public async handleGetFriendslist(req: express.Request, res: express.Response) {
+    const { userId } = req.session;
+
+    try {
+      return await this.controller.getFriendslist(userId!);
+    } catch (err) {
+      return new Error('Internal Server Error');
+    }
+  }
+
+  public async handleGetFriendRequests(req: express.Request, res: express.Response) {
+    const { userId } = req.session;
+
+    try {
+      const friendRequestsResult = await this.controller.getFriendRequests(userId!);
+      return friendRequestsResult;
+    } catch (err) {
+      return new Error('Internal Server Error');
+    }
+  }
+
+  public async handleCreateFriendRequest(req: express.Request, res: express.Response) {
+    const { userId: inviterId } = req.session;
+    const { message } = req.body;
+    const inviteeId = parseInt(req.params.userId, 10);
+
+    if (!inviteeId) {
+      return new Error('Must provide a valid user id');
+    }
+
+    if (inviteeId === inviterId) {
+      return new Error('You cannot send a friend request to yourself');
+    }
+
+    // TODO: different responses depending on the status of the friendship
+    const existingFriendship = await this.controller.getFriendshipIds(inviterId!, ['accepted', 'pending', 'blocked', 'rejected']);
+    if (existingFriendship.includes(inviteeId)) {
+      return new Error('You are already have a relationship with this user');
+    }
```

Comment:
There are 8 kinds of relationships to consider (4 statuses, and user could be inviter or invitee). Ain't got time for that. I *think* this should suffice.